modules/serverless-creating-broker.adoc

```diff
@@ -0,0 +1,63 @@
+// Module included in the following assemblies:
+//
+// * serverless-using-brokers.adoc
+
+[id="serverless-creating-broker_{context}"]
+= Creating a broker
+
+To create a broker, you must create a `ServiceAccount` for each namespace and give that `ServiceAccount` the required RBAC permissions.
+
+.Prerequisites
+* Knative Eventing installed, which includes the `ClusterRole`.
+
+.Procedure
+. Create the `ServiceAccount` objects:
++
+----
+$ oc -n default create serviceaccount eventing-broker-ingress
+$ oc -n default create serviceaccount eventing-broker-filter
+----
+
+. Give those objects RBAC permissions:
++
+----
+$ oc -n default create rolebinding eventing-broker-ingress \
+  --clusterrole=eventing-broker-ingress \
+  --serviceaccount=default:eventing-broker-ingress
+$ oc -n default create rolebinding eventing-broker-filter \
+  --clusterrole=eventing-broker-filter \
+  --serviceaccount=default:eventing-broker-filter
+----
+
+. Create RBAC permissions granting access to shared configmaps for logging, tracing, and metrics configuration (optional):
```

Comment:
@matzew  ^^ Are these supposed to be optional? Without those rolebindings, the default-broker-filter and ingress crashloops with

2020/02/14 15:56:10 Error loading/parsing logging configuration:configmaps "config-logging" is forbidden: User "system:serviceaccount:manual:eventing-broker-filter" cannot get resource "configmaps" in API group "" in the namespace "knative-eventing"

(with quay.io/openshift-knative/knative-eventing-filter:v0.12.0 and quay.io/openshift-knative/knative-eventing-ingress:v0.12.0 )

Reply:
hrm... I am not sure why the doc says optional.

yoiu can either do injection fu, or do it manually  

upstream has some doc on this: knative-eventing create rolebinding eventing-config-reader-default-eventing-broker-ingr

Reply:
Also, have we forgotten about this one? https://github.com/knative/eventing/pull/2034   (this is supposed to create the default broker just by having a trigger with "knative-eventing-injection: enabled" annotation.

Reply:
@maschmid yep, it's lacking in the upstream docs.

I did add a PR here: https://github.com/knative/docs/pull/2285

Reply:
> Are these below supposed to be optional? 

I just tested 0.13.0 on minikube, and it works without the rolebindings there .

Next: testing OCP -> migjt be possible, that they are required there...

Reply:
@abrennan89 @maschmid  using 0.13 on OCP 4.3  this works too!

the rolebindings _ARE_ optional.

i did not create them, and was able to get broker/trigger (v1betaa1) working

Reply:
It still doesn't work for me without the rolebindings (using v0.13.0 from https://github.com/openshift/knative-eventing/blob/release-v0.13.0/openshift/olm/knative-eventing.catalogsource.yaml  )

default-broker-filter and default-broker-ingress still crashloops

`2020/03/13 12:09:17 Error loading/parsing logging configuration:configmaps "eventing-config-logging" is forbidden: User "system:serviceaccount:user2:eventing-broker-ingress" cannot get resource "configmaps" in API group "" in the namespace "user2"`

Reply:
Actually it doesn't even work if I do create them... (because in this case it needs to be able to read the configmap not from knative-eventing, but from the users' namespace  (in my case "user2", "default" in this example)

Reply:
nm, I have wrongly translated the "default" above... works fine now!